### PR TITLE
os-specific/linux: replace name with pname&version

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Ericson2314 — APPROVED** [review]

**Review by r-burns — APPROVED** [review]
Yeah I think the check error message is misleading. `cachix: CppStdException "Exception: path '/nix/store/rl1bm7f6z36yvjrmwny1wz4lrnq1nm9i-python3.9-watchdog-2.1.4' is not valid; type: nix::InvalidPath"` happened because python3.pkgs.watchdog failed to build as a dependency of the manual. (Might have run out of resources or timed out)

However python3.pkgs.watchdog builds locally for me, so LGTM.

File: pkgs/os-specific/linux/fswebcam/default.nix [hunk 1]
```diff
@@ -1,10 +1,11 @@
 { lib, stdenv, fetchurl, libv4l, gd }:
 
 stdenv.mkDerivation rec {
-  name = "fswebcam-20200725";
+  pname = "fswebcam";
+  version = "2020-07-25";
 
   src = fetchurl {
-    url = "https://www.sanslogic.co.uk/fswebcam/files/${name}.tar.gz";
+    url = "https://www.sanslogic.co.uk/fswebcam/files/${lib.replaceStrings ["."] [""] version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/fswebcam/default.nix:8]:
```suggestion
    url = "https://www.sanslogic.co.uk/fswebcam/files/fswebcam-${lib.replaceStrings ["."] [""] version}.tar.gz";
```

**Review by Ericson2314 — CHANGES_REQUESTED** [review]
Ah good catch @SuperSandro2000

File: pkgs/os-specific/linux/afuse/default.nix [hunk 2]
```diff
@@ -1,10 +1,11 @@
 { lib, stdenv, fetchurl, pkg-config, autoreconfHook, fuse }:
 
-stdenv.mkDerivation {
-  name = "afuse-0.4.1";
+stdenv.mkDerivation rec {
+  pname = "afuse";
+  version = "0.4.1";
 
   src = fetchurl {
```
Comment (Mic92) [inline pkgs/os-specific/linux/afuse/default.nix:7]: Look like this predates `fetchFromGithHub`

File: pkgs/os-specific/linux/afuse/default.nix [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+  version = "0.4.1";
 
-  src = fetchurl {
-    url = "https://github.com/pcarrier/afuse/archive/v0.4.1.tar.gz";
-    sha256 = "1sfhicmxppkvdd4z9klfn63snb71gr9hff6xij1gzk94xg6m0ycc";
+  src = fetchFromGitHub {
+    owner = "pcarrier";
+     repo = "afuse";
+    rev = "v${version}";
+    sha256 = "06i855h8a1w2jfly2gfy7vwhb2fp74yxbf3r69s28lki2kzwjar6";
```
Comment (bobby285271) [inline pkgs/os-specific/linux/afuse/default.nix:11]: Nit: indent.

```suggestion
    owner = "pcarrier";
    repo = "afuse";
    rev = "v${version}";
    sha256 = "06i855h8a1w2jfly2gfy7vwhb2fp74yxbf3r69s28lki2kzwjar6";
```

**Pull request merged**